### ISSUE TITLE
bad assumptions on os_type population 

### ISSUE THREAD

**Issue by chorankates-sfdc** [chorankates/rouster#88]
```rb
NoMethodError: undefined method `keys' for nil:NilClass
    ~/rouster/lib/rouster/deltas.rb:482:in `get_services'
    ~/rouster/lib/rouster/testing.rb:628:in `validate_service'
```

`get_services` calls `os_type` to determine what OS it is on so appropriate commands can be run, but if none of the flag files that we look for exist, and `type=:all`, we end up with this exception.

in `get_packages`, an internal error is thrown if we don't recognize the OS, same with `get_ports`, so going to make this uniform

**Comment by chorankates-sfdc** [comment]
proposed fix in os_type-population branch

**Comment by chorankates-sfdc** [comment]
resolved in #90
